lib/htmlbars-plugin/index.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+var utils = require('./utils');
+
+function Plugin() {
+  this.syntax = null;
+  this.builders = null;
+}
+
+Plugin.prototype.constructor = Plugin;
+
+Plugin.prototype.transform = function(ast) {
+  if (!this.builders) {
+    this.builders = this.syntax.builders;
+  }
+
+  var plugin = this;
+  var walker = new this.syntax.Walker();
+
+  walker.visit(ast, function(node) {
+    if (plugin.validate(node)) {
+      switch(node.type) {
+        case 'ElementNode':
+          plugin.parseElementNode(node);
+          break;
+        case 'MustacheStatement':
+        case 'BlockStatement':
+          plugin.parseStatement(node);
+          break;
+      }
+    }
+  });
+
+  return ast;
+};
+
+Plugin.prototype.validate = function(node) {
+  return ['ElementNode', 'MustacheStatement', 'BlockStatement'].indexOf(node.type) > -1;
+};
+
+Plugin.prototype.parseStatement = function(node) {
+  var localClassPair = utils.getPair(node, 'local-class');
+
+  if (!localClassPair) { return; }
+
+  utils.removePair(node, localClassPair);
+
+  var classPair = utils.getPair(node, 'class');
+  var params = [];
+
+  if (classPair) {
+    params.push(classPair.value);
+    params.push(this.seperator());
+    utils.removePair(node, classPair);
+  }
+
+  var concatSexpr = this.builders.sexpr('concat', params);
+  node.hash.pairs.push(this.builders.pair('class', concatSexpr));
+  this.setLocalClasses(params, localClassPair.value.value);
+};
+
+Plugin.prototype.parseElementNode = function(node) {
+  var localClassAttr = utils.getAttr(node, 'local-class');
+
+  if (!localClassAttr) { return; }
+
+  utils.removeAttr(node, localClassAttr);
+
+  var classAttr = utils.getAttr(node, 'class');
+  var parts = [];
+
+  if (classAttr) {
+    utils.removeAttr(node, classAttr);
+
+    var classAttrValue = classAttr.value;
+
+    // Unwrap the original class attribute value
+    if (classAttrValue.type === 'ConcatStatement') {
```

Comment:
78-88 I'm unsure if there is a better way to do this.  Basically I need to concat the existing class attribute values with the replacement class attribute.